### PR TITLE
Traitor Values adjustment

### PR DESCRIPTION
**Pull request by Ranzarth**
Adjust the amount of traitors that spawn for balancing
Also readds a check for the second return to stop duplications

### CODE REVIEW

**Comment by daalta** [comment]
Looks OK to merge! Having 4 traitors at ten players seems silly, your numbers are much better.

50+ players: 5 traitors
30+ players: 4 traitors
15+ players: 2 traitor
00+ players: 1 traitor

What do you think of adding a player count at which 3 traitors spawn? At, let's say, 23 players?

**Comment by daalta** [comment]
I've gone ahead and simplified the check. https://github.com/unitystation/unitystation/pull/3761/commits/6e66e4c933ddc3ac6df0d88515761cf49a13f05b

**Comment by corp-0** [comment]
make it a switch! We don't want our code to look like Yandere Simulator

**Comment by daalta** [comment]
@corp-0 A switch would be better... but eh, I think it's not worth the effort. Code won't be any shorter, and the variable is only referenced thrice. Good enough! :)

**Comment by daalta** [comment]
Added 3 traitors when there are at least 24 players, discussed it with @Ranzarth .

**Comment by daalta** [comment]
This doesn't really make sense to me - why is this a for loop in the first place? https://github.com/unitystation/unitystation/pull/3761/commits/c3f5575d843ac551bb959c5fc57e087d739d602d

~~Either way, I think this can be merged.~~ Uh, nevermind

`AntagManager.Instance.AntagCount == TraitorAmount`

This part eludes me. Should is be less / more or equal than?

Maybe someone else can pick up the review o.o

File: UnityProject/Assets/Scripts/GameModes/Traitor.cs [hunk 1]
```diff
@@ -68,11 +76,12 @@ public override void StartRound()
...
 	{
 
-		for (int i = 0; i < TraitorAmount - 1; i++) {
+		for (int i = 1; i < TraitorAmount; i++) {
 			return !LoyalImplanted.Contains(spawnRequest.RequestedOccupation.JobType)
 					&& PlayerList.Instance.InGamePlayers.Count > 0;
 		}
 		return !LoyalImplanted.Contains(spawnRequest.RequestedOccupation.JobType)
+					&& AntagManager.Instance.AntagCount == 0
 					&& PlayerList.Instance.InGamePlayers.Count > 0;
```
Comment (daalta) [inline UnityProject/Assets/Scripts/GameModes/Traitor.cs]: Why the for loop? Why not replace
```
for (int i = 1; i < TraitorAmount; i++) {
	return !LoyalImplanted.Contains(spawnRequest.RequestedOccupation.JobType)
			&& PlayerList.Instance.InGamePlayers.Count > 0;
}
return !LoyalImplanted.Contains(spawnRequest.RequestedOccupation.JobType)
			&& AntagManager.Instance.AntagCount == 0
			&& PlayerList.Instance.InGamePlayers.Count > 0;
```
with
```
return !LoyalImplanted.Contains(spawnRequest.RequestedOccupation.JobType)
			&& AntagManager.Instance.AntagCount <= TraitorAmount
			&& PlayerList.Instance.InGamePlayers.Count > 0;
```

Comment (Ranzarth) [reply UnityProject/Assets/Scripts/GameModes/Traitor.cs]: Hmm I didn't really take that into account

**Review by Jibbajabbafic — CHANGES_REQUESTED** [review]
You can remove TraitorAmount all together and replace it with a ratio instead. See `NukeOps.cs` for an example.

**Comment by Jibbajabbafic** [comment]
Nice, the ratio might need adjusting to find a good balance but we can figure that out later on

**Review by Jibbajabbafic — APPROVED** [review]

**Comment by Ranzarth** [comment]
Indeed, I just set it to the same value as nukeops for now

**Comment by Jibbajabbafic** [comment]
I'll let @daalta decide if he wants to merge this as is

**Pull request merged**